### PR TITLE
Milestone 4 : OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by ABDULALI3468**
> # What I did ♻️
- In this project, I created a form of UI for your school library app. This way it can be invoked as an executable and not something you use in IRB exclusively.

> ## Project requirements ♻️
- Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
    - [UI example](https://www.youtube.com/watch?v=vkkgrhD6aXQ)
- Your console app should behave in the same way as it is in the example.
- Create a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain `Person`).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person id.
- In your `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method should do the following:
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        app.print_person
+      when 3
+        app.add_person
+      when 4
+        app.create_book
+      when 5
+        app.create_rental
+      when 6
+        app.list_rentals
+        break
```
Comment (zainsadaqat) [inline main.rb]: I'm trying to exit from the project but unable to do so as it's missing the exit point. Please add an exit case as well. 👍🏻

**Review by zainsadaqat — CHANGES_REQUESTED** [review]
# Required Changes 🔁
# Hello @ABDULALI3468  😄 

You're doing a great job on your project. We're proud of you. Good job so far! 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
  ✅  Correct use of Github Flow
  ✅  No linter errors
  ✅  Descriptive Pull Request

## Required Changes ♻️

_Check the comments under the review._




## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (ABDULALI3468) [reply main.rb]: @zainsadaqat 
 😵‍💫 no problem I will do that. Thank you so much for your review. Can you star my projects and do interact with me on the following platforms.

👤 **ABDUL ALI**

> * This is My **GitHub**: GitHub followers
> * This is My **Linkedin**: [Linkedin Badge](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [Twitter Badge](https://twitter.com/mrabdul_ali)&nbsp;

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  private :of_age?
+end
+
+# person = Person.new(22, 'maximilianus')
+# person.correct_name
+# capitalized_person = CapitalizeDecorator.new(person)
+# puts capitalized_person.correct_name
+# capitalized_trimmed_person = TrimmerDecorator.new(capitalizedPerson)
+# puts capitalized_trimmed_person.correct_name
```
Comment (CollinsTatang) [inline person.rb:40]: - [ ] **Optional**, Please remove all commented code from your codebase to make it clean and easy to maintain.

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**